tests/unit/customizations/s3/test_s3handler.py

```diff
@@ -914,6 +915,25 @@ def test_upload_stream_with_expected_size(self):
         # New chunksize should have a total parts under 1000.
         self.assertTrue(100000 / float(changed_chunk_size) <= MAX_PARTS)
 
+    def test_upload_stream_with_expected_size(self):
+        minimum_chunksize = 10
+        awscli.customizations.s3.utils.MIN_UPLOAD_CHUNKSIZE = minimum_chunksize
+        s3handler = S3StreamHandler(
+            self.session, self.params,
+            runtime_config=runtime_config(
+                multipart_threshold=1, multipart_chunksize=2))
+        s3handler.executor = mock.Mock()
+        fileinfo = FileInfo('filename', operation_name='upload',
+                            is_stream=True)
+        with MockStdIn(b'bar'):
+            s3handler._enqueue_multipart_upload_tasks(fileinfo, b'')
+        submitted_tasks = s3handler.executor.submit.call_args_list
+        # Determine what the chunksize was changed to from one of the
+        # UploadPartTasks.
+        changed_chunk_size = submitted_tasks[1][0][0]._chunk_size
```

Comment:
Not a fan of accessing the internal ``_chunk_size``. Not sure there is a great way to grab this though without large amounts of rewrites...